rs_client.py

```diff
@@ -1019,16 +1019,17 @@ def _rs_struct(typeobj):
 
         _r('pub type %s%s = base::StructPtr<%s%s>;', typeobj.rs_type, lifetime1,
                 lifetime2, typeobj.ffi_type)
-        typeobj.struct_ptr = True
 
 
 def _rs_accessors(typeobj):
 
     lifetime = "<'a>" if typeobj.has_lifetime else ""
+    # unions always get an impl lifetime to be sure
+    lifetime2 = "<'a>" if typeobj.is_union else lifetime
 
```

Comment:
That works as well. I've built the code in the assumption that it's probably easier to define the lifetime once (easier as in, easier to read the code we generate). However, defining the lifetime for each accessor function would allow for a cleaner codegen solution. The two are equivalent in functionality, however. Do you prefer the latter option?